### ISSUE TITLE
wb_save outputs a file which Excel perceives as corrupted

### ISSUE THREAD

**Issue by itsmevictor** [JanMarvin/openxlsx2#1032]
My issue is that regardless of the content of the workbook and regardless of the modifications I perform on it, the file outputted by `wb_save` is perceived as corrupted by Excel (but not Libre Office). 

I ended up testing with an empty .xlsx file containing only the value 1 in the cell A1 and ran the following (that is without performing any modifications on the workbook): 
```
template_file <- openxlsx2::wb_load(file = paste0(wd, "/data/template_liste.xlsx"))

openxlsx2::wb_save(wb = template_file, file = paste0(wd, "/data/test_template_liste.xlsx"), overwrite = TRUE)
```
but for some reason, when trying to open the file in Excel, I get the following error message _"We found a problem with some content in 'test_template_liste.xlsx'. Do you want us to try to recover as much as we can? If you trust the source of this workbook, click Yes."._. If I click Yes, the content of the repair is the following:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
> <recoveryLog xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main"><logFileName>Repair Result to test_template_liste6.xml</logFileName><summary>Errors were detected in file '/data/test_template_liste.xlsx'</summary><additionalInfo><info>Excel completed file level validation and repair. Some parts of this workbook may have been repaired or discarded.</info></additionalInfo><repairedParts summary="Following is a list of repairs:"><repairedPart>Repaired Part: /xl/workbook.xml part with XML error.  (Workbook) Xml parsing error Line 1, column 1679.</repairedPart></repairedParts></recoveryLog>
```

This repair message appears to be the same regardless of the content of the workbook. Noteworthily, I do not have the same issue with LibreOffice *and* I do not have the same issue with Python's openpyxl (including for Excel). 

I'm using Excel 16.85 on Mac OS and I'm using R 4.4.0 in Visual Studio Code 1.89.1. An example of a corrupted Excel file is here: 
[test_template_liste.xlsx](https://github.com/JanMarvin/openxlsx2/files/15501830/test_template_liste.xlsx)

I hope the issue is clear enough. Many thanks in advance for your help!

**Comment by JanMarvin** [comment]
Hi @itsmevictor , Thanks for the report, I’ll look into it.

Two things
* Can you please provide the file prior to our interaction with it? The file you want to load and save.
* Does it work if you save a file not loaded? Like `write_xlsx()`

**Comment by JanMarvin** [comment]
This file has a reference to something external, but `rId6` is missing. Either lost when saving in LibreOffice, loading the file in openxlsx2 or something we do not (yet) process. In the file you have provided, there simply is nothing related to this:

```XML
<extLst>
  <ext uri="GoogleSheetsCustomDataVersion2">
    <go:sheetsCustomData xmlns:go="http://customooxmlschemas.google.com/" r:id="rId6" .../>
  </ext>
</extLst>
```

You can remove this field and the file opens without any error:
```R
library(openxlsx2)
wb <- wb_load("~/Downloads/test_template_liste.xlsx")
wb$workbook$extLst <- character()
if (interactive()) wb$open()
```

**Comment by itsmevictor** [comment]
Hi @JanMarvin, many thanks for your quick response! To answer your first comment, here is the file prior to my interaction with it: 
[template_liste.xlsx](https://github.com/user-attachments/files/15509477/template_liste.xlsx)

`write_xlsx` works with no problem. Running 
```tibble(x = 1) |> openxlsx2::write_xlsx(file = paste0(wd, "/data/test_template_liste3.xlsx"))``` 
creates an .xlsx file which Excel perceives as not corrupted. 

To answer your second point, I'm not entirely sure I made myself clear: my problem is that the file saved by `wb_save` appears corrupted when I try to open it in Excel, not when I try to process it in R. 

Let me know if I can provide any additional clarification!

**Comment by JanMarvin** [comment]
Thanks, your explanation is perfectly fine. The issue stems from an internal binary file `metadata` in the `xl` folder in your xlsx file. This binary file looks like it belongs to a custom extension (custom as in it’s not part of the ooxml standard). Hence the question is, do you need this file and is it expected? I assume that the binary file belongs to google spreadsheets and is used for data synchronization.

I have a draft branch that handles this worksheet (I’ll push it after some further cleanup), but I’m still a bit in doubt if I want to blindly handle binary blobs.

**Comment by itsmevictor** [comment]
Thanks for the clarification. I think you're right. 

I happen to use Google Drive for Mac (previously Backup and Sync) so I'm thinking it might be linked to that, with Google Drive adding this binary file to facilitate synchronization. 

I just tested the initial calls mentioned in the initial post **outside of a folder synced with Google Drive** and the result is the following:
- If the loaded file was created **outside** of a Google Drive synced folder, the file outputted by `wb_save` is **not corrupted**.
- If the loaded file was created **within** a Google Drive synced folder, the file outputted by `wb_save` **is corrupted**, even if the file is now located outside a Google Drive synced folder. 

So I think this confirms your intuition, being that Google Drive performs some weird stuff on a file as soon as it's stored in a Google Drive synced folder (which stay with the file even if the file is moved), and this is what is causing the issue. 

Let me know if that does not clarify any remaining doubts. The question of whether blindly allowing binary blobs is appropriate or not goes beyond my technical skills and I trust that you will choose what's best, but I'm happy to help debugging/testing things if necessary! 

Cheers, and thank you again both for the great package and for your help :-)

**Comment by JanMarvin** [comment]
Hi @itsmevictor , I have a PR open in #1033. You should be able to install from this branch with:

```R
remotes::install_github("JanMarvin/openxlsx2#1033")
```

This branch detects the `xl/metadata` file and keeps it. This way the file exported with `wb_save()` looks identical to the one that you imported. It would be great if you can check if this branch works for you. Since I don't know how the internal binary blob is used, and I was unable to find any documentation, keeping it might cause other unintended problems. Syncing with the cloud might be affected: either the file is not synced anymore, because the local software assumes that the file is still in sync or that the file in the cloud is always up to date.

I simply don't know, and files like this are called binary blobs for a reason. Hence my initial approach was to remove the remaining reference and ignore whatever the google software put inside the xlsx file (this is what I did here: https://github.com/JanMarvin/openxlsx2/issues/1032#issuecomment-2140962241)

> Cheers, and thank you again both for the great package and for your help :-)

Thank you!

**Comment by itsmevictor** [comment]
Hi @JanMarvin! 

I just tested after installing #1033 and it works perfectly! Both the inputted and outputted file sync perfectly with Google Drive, and the outputted file is not anymore recognized by Excel as corrupted. 

Wonderful work, thank you again :-) 
Have a good weekend!

**Comment by JanMarvin** [comment]
Thank you for the test. I would like to try something else because I am not entirely satisfied with the previous approach. In #1034 I have prepared another PR that simply removes the binary blob and its remaining reference from the xlsx file. Could you please try this branch as well? It specifically searches for the Google Sheets extension and removes the reference. The PR is much smaller because we don't have to jump through various loops to create the entire required xml structure.

**Comment by JanMarvin** [comment]
I’ve taken another approach. Should be fixed with 1.7.
